abci/client/client.go

```diff
@@ -33,35 +33,24 @@ type Client interface {
 
 	// Asynchronous requests
 	FlushAsync(context.Context) (*ReqRes, error)
```

Comment:
Why do we still need this?

Reply:
We might not. It's called in the [mempool](https://github.com/tendermint/tendermint/blob/mjf/desync/internal/mempool/mempool.go#L738) and the [proxy app](https://github.com/tendermint/tendermint/blob/mjf/desync/internal/proxy/app_conn.go#L142). It might be we could get rid of those, or swap them to synchronous, but I didn't want to expand this change with functional changes.

Reply:
Yeah makes sense to tackle this as a different issue/PR